Tools/lib/python/eupath/EupathExporter.py

```diff
@@ -6,387 +6,229 @@
-
-        if self._validation_script == None:
-            return
-        
-        dataset_files = self.identify_dataset_files()
-
-        validation_process = Popen(['python', self._tool_directory + "/../../bin/" + self._validation_script],
-                                   stdin=PIPE, stdout=PIPE, stderr=PIPE)
-        # output is a tuple containing (stdout, stderr)
-        output = validation_process.communicate(json.dumps(dataset_files).encode("utf-8"))
-        if validation_process.returncode == 1:
-            raise ValidationException(output[1])
-
-    def identify_dependencies(self):
-        """
-        An abstract method to be addressed by a specialized export tool that furnishes a dependency json list.
-        :return: The dependency json list to be returned should look as follows:
-        [dependency1, dependency2, ... ]
-        where each dependency is written as a json object as follows:
-        {
-          "resourceIdentifier": <value>,
-          "resourceVersion": <value>,
-          "resourceDisplayName": <value
-        }
-        Where no dependencies exist, an empty list is returned
-        """
-        raise NotImplementedError(
-            "The method 'identify_dependencies(self)' needs to be implemented in the specialized export module.")
-
-    def identify_projects(self):
-        """
-        An abstract method to be addressed by a specialized export tool that furnishes a VEuPathDB project list.
-        :return: The project list to be returned should look as follows:
-        [project1, project2, ... ]
-        At least one valid VEuPathDB project must be listed
-        """
-        raise NotImplementedError(
-            "The method 'identify_project(self)' needs to be implemented in the specialized export module.")
-
-    def identify_supported_projects(self):
-        """
-        Override this method to provide a non-default list of projects.
+            return (config_json["url"], config_json["user"], config_json["password"], config_json["service-url"], config_json["super-user-token"])
+                    
+    def export(self):
 
-        Default is None, interpreted as all projects are ok, ie, no constraints.
-        """
-        return None;
+        # Save the current working directory so we can get back to it
+        orig_path = os.getcwd()
 
+        with self.temporary_directory(self._export_file_root) as temp_path:
+            os.chdir(temp_path)
+            print_debug("temp path: " + temp_path)
+            self.prepare_data_files(temp_path)
+            tarball_name = self.create_tarball(temp_path)
+            json_body = self.create_body_for_post()
+            print_debug(json_body)
+            user_dataset_id = self.post_metadata_json(json_body)
+            print_debug("UD ID: " + user_dataset_id)
+            self.post_datafile(user_dataset_id, tarball_name)
+            self.poll_for_upload_complete(user_dataset_id)   # teriminates if system or validation error
```

Comment:
What's the UX if this throws?

Reply:
in Galaxy, when a processing step fails, the user is shown stdout/stderr